tests/net/socket/tcp/prj.conf

```diff
@@ -26,10 +26,8 @@ CONFIG_NET_CONFIG_MY_IPV6_ADDR="2001:db8::1"
 
 CONFIG_MAIN_STACK_SIZE=2048
 
-# Required:
-# Net pkt will be reused since src and dst address are the same.
-# It takes at least 3 tx pkt to establish TCP connection (syn/syn-ack/ack)
-CONFIG_NET_PKT_TX_COUNT=6
+# The test requires lot of bufs
+CONFIG_NET_PKT_TX_COUNT=24
```

Comment:
This is quite an increase... Maybe it's a sign that we have more buffer leaks somewhere?